src/NuGetPackageVerifier/Rules/PackageVersionMatchesAssemblyVersionRule.cs

```diff
@@ -6,39 +6,54 @@
 using System.Linq;
 using System.Reflection;
 using Mono.Cecil;
+using NuGet.Versioning;
 
 namespace NuGetPackageVerifier.Rules
 {
     public class PackageVersionMatchesAssemblyVersionRule : AssemblyHasAttributeRuleBase
     {
-        private Version _packageVersion;
+        private bool _isPrerelease;
+        private NuGetVersion _packageVersion;
         private string _packageId;
 
         public override IEnumerable<PackageVerifierIssue> Validate(PackageAnalysisContext context)
         {
-            _packageVersion = context.Metadata.Version.Version;
+            _isPrerelease = context.Metadata.Version.IsPrerelease;
+            _packageVersion = context.Metadata.Version;
             _packageId = context.Metadata.Id;
             return base.Validate(context);
         }
 
         public override IEnumerable<PackageVerifierIssue> ValidateAttribute(string currentFilePath,
             Mono.Collections.Generic.Collection<CustomAttribute> assemblyAttributes)
         {
+
             var versionAttribute = assemblyAttributes.SingleOrDefault(a =>
                 a.AttributeType.FullName.Equals(
-                    typeof(AssemblyInformationalVersionAttribute).FullName ??
-                    typeof(AssemblyVersionAttribute).FullName,
+                    typeof(AssemblyInformationalVersionAttribute).FullName,
                     StringComparison.Ordinal));
 
             if (versionAttribute == null)
             {
                 yield break;
             }
 
-            var assemblyVersion = versionAttribute.AttributeType.Module.Assembly.Name.Version;
-            if (!_packageVersion.Equals(assemblyVersion))
+            if (_isPrerelease)
```

Comment:
See comment above, we always need to check both info version and real version. The check does need to vary based on IsPreRelease (because real version is never pre-release), but both attributes always need to be checked.